### PR TITLE
Fixes for genetlink

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: dataplane/forwarding/fwdport/ports/cpu.go [hunk 1]
```diff
@@ -131,97 +108,30 @@ func (p *CPUPort) Write(packet fwdpacket.Packet) (fwdaction.State, error) {
...
-	egressID, err := strconv.Atoi(egressPID.GetObjectId().GetId())
-	if err != nil {
-		return
-	}
 
 	response := &pktiopb.PacketOut{
 		Packet: &pktiopb.Packet{
 			InputPort:  uint64(ingressID),
-			OutputPort: uint64(egressID),
+			OutputPort: uint64(0),
```
Comment (guoshiuan) [inline dataplane/forwarding/fwdport/ports/cpu.go:134]: Can you add a comment on why 0 (previously egressID) is used?

File: dataplane/forwarding/fwdport/ports/cpu.go [hunk 2]
```diff
@@ -234,8 +144,8 @@ func (p *CPUPort) puntRemotePort(v any) {
 		return
 	}
 
-	timer := deadlock.NewTimer(deadlock.Timeout, fmt.Sprintf("Punting packet from port %v", p))
-	defer timer.Stop()
+	// timer := deadlock.NewTimer(deadlock.Timeout, fmt.Sprintf("Punting packet from port %v", p))
```
Comment (guoshiuan) [inline dataplane/forwarding/fwdport/ports/cpu.go:147]: If the code is not use, please delete them.

File: dataplane/forwarding/fwdport/ports/cpu_test.go [hunk 3]
```diff
@@ -34,46 +36,34 @@ import (
...
-	name := "TestPort"
+	name := "1"
 	ctx := fwdcontext.New("test", "fwd")
 	ps := &RecordPacketSink{
 		notify: make(chan bool),
 	}
-	ctx.SetPacketSink(ps.PacketSink)
+	ctx.SetCPUPortSink(ps.Send, func() {})
 
 	// Create a CPU port that exports the ETHER_TYPE and IP_ADDR_DST
```
Comment (guoshiuan) [inline dataplane/forwarding/fwdport/ports/cpu_test.go:59]: Revise the comment?

File: dataplane/kernel/genetlink/genetlink.c [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  struct nl_sock* sock = nl_socket_alloc();
+  if (sock == NULL) {
+    fprintf(stderr, "error: failed to alloc nl socket");
+    return NULL;
+  }
+  nl_socket_disable_auto_ack(sock);
+  int error = genl_connect(sock);
+  if (error < 0) {
+    fprintf(stderr, "error: failed to disable auto ack");
```
Comment (guoshiuan) [inline dataplane/kernel/genetlink/genetlink.c:44]: Do we want to also print the error code?

File: dataplane/kernel/genetlink/genetlink.c [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  nl_socket_set_peer_groups(sock, (1 << (group_id - 1)));
+  return sock;
+}
+
+void delete_port(void* sock) { nl_socket_free(sock); }
+
+int send_packet(void* sock, int family, const void* pkt, uint32_t size,
+                int in_ifindex, int out_ifindex, unsigned int context) {
+  struct nl_msg* msg = nlmsg_alloc();
+  genlmsg_put(msg, NL_AUTO_PORT, NL_AUTO_SEQ, family, 0, 0, 0, 1);
```
Comment (guoshiuan) [inline dataplane/kernel/genetlink/genetlink.c:63]: Please make sure msg is not NULL.

**Review by guoshiuan — APPROVED** [review]

File: dataplane/forwarding/fwdport/ports/cpu.go (hunk 1 above)
Comment (DanG100) [reply dataplane/forwarding/fwdport/ports/cpu.go]: added todo

File: dataplane/kernel/genetlink/genetlink.c (hunk 4 above)
Comment (DanG100) [reply dataplane/kernel/genetlink/genetlink.c]: done

File: dataplane/kernel/genetlink/genetlink.c (hunk 5 above)
Comment (DanG100) [reply dataplane/kernel/genetlink/genetlink.c:67]: done

**Pull request merged**